### PR TITLE
fix: Connected on Polkadot, opening Base link on Koda, mint ing - starts loading

### LINKED ISSUE #10753: Connected on Polkadot, opening Base link on Koda, minting - starts loading
Problem
---
Now Im logged in with Polkadot wallet. I clicked on this link that I received in my message: https://kodadot.xyz/base/drops/rvn3. (It's a link to the Base drop page). So I'm logged in with Polkadot wallet now, and I have Base Drop in front of me. 

And I clicked on MINT - it started loading (doing nothing but loading). From what I remember, wasn't I supposed to get popup to switch wallet to EVM? 

Video
---

https://github.com/user-attachments/assets/72e5c072-b07b-41a5-8dd5-202a91cea387

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [Koda - Generative Art Marketplace](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Closes #10753

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

image

### CODE REVIEW

**Review by preschian — APPROVED** [review]

**Comment by vikiival** [comment]
Screenshot 2024-08-08 at 17 35 21

@JustLuuuu is OOO 
- https://github.com/kodadot/ops-internal/discussions/1781#discussioncomment-10242554

**Review by hassnian — DISMISSED** [review]
small, otherwise LGTM

File: stores/wallet.ts [hunk 1]
```diff
@@ -38,6 +39,7 @@ export const useWalletStore = defineStore('wallet', {
     disconnecting: false,
   }),
   getters: {
+    getWalletVM: state => state.selected?.vm || 'SUB',
```
Comment (hassnian) [inline stores/wallet.ts:42]: actually this is not accurate , you might be logged out , and on a evm drop and this would return 'SUB'

```suggestion
    getWalletVM: state => state.selected?.vm,
```

CleanShot 2024-08-09 at 10 50 08@2x

File: components/collection/drop/PaidGenerative.vue [hunk 2]
```diff
@@ -104,6 +107,11 @@ const handleSubmitMint = async () => {
     return false
   }
 
+  if (vm.value !== getWalletVM.value) {
```
Comment (hassnian) [inline components/collection/drop/PaidGenerative.vue:110]: let's make it a getter , called `getIsVmChain` so we can resuse it

**Review by hassnian — CHANGES_REQUESTED** [review]

**Comment by vikiival** [comment]
cc @Jarsen136

File: stores/wallet.ts [hunk 3]
```diff
@@ -2,6 +2,7 @@ import { defineStore } from 'pinia'
 import type { ChainVM, Prefix } from '@kodadot1/static'
 import { isDateWithinLastDays } from '@/utils/datetime'
 import { ss58Of } from '@/utils/config/chain.config'
+import { formatAddress } from '@/utils/account'
```
[the remark this answers is not in the document]
Comment (Jarsen136) [reply stores/wallet.ts:5]: It should work with auto-import, but it does not work somehow on my localhost env.

**Comment by Jarsen136** [comment]
All should have been updated. May I get your code review again? @hassnian

**Comment by Jarsen136** [comment]
> > All should have been updated. May I get your code review again? @hassnian
> 
> Could you do [#10772 (comment)](https://github.com/kodadot/nft-gallery/pull/10772#discussion_r1709143340)
> 
> This way 'useDoAfterReconnect' takes at least on success and we can have open 'openReconnectModal' without anything passed like https://github.com/Jarsen136/nft-gallery/blob/issue-10753/composables/useDoAfterlogin.ts
> 
> Tyvm

Updated

**Comment by hassnian** [comment]
@Jarsen136 did you push your changes ?

CleanShot 2024-08-11 at 12 39 29@2x

**Comment by Jarsen136** [comment]
> related?
> 
> CleanShot 2024-08-11 at 12 40 30 

Yes, there is something wrong on GitHub I guess. Let me try to push another commit.

**Comment by hassnian** [comment]
@Jarsen136 dont remove the useReconnectModal use the openreconnectmodal there

**Comment by Jarsen136** [comment]
> @Jarsen136 dont remove the useReconnectModal use the openreconnectmodal there


Do you mean like this?
```
const { openReconnectWalletModal } = useReconnectModal()
```

I don't get the benefit or major difference of using it as a composable instead of utils function.
Could you explain a bit? In the previous implementation, it was a composable function but with a different name.

**Comment by hassnian** [comment]
> > @Jarsen136 dont remove the useReconnectModal use the openreconnectmodal there
> 
> Do you mean like this?
> 
> ```
> const { openReconnectWalletModal } = useReconnectModal()
> ```
> 
> I don't get the benefit or major difference of using it as a composable instead of utils function. Could you explain a bit? In the previous implementation, it was a composable function but with a different name.

sure, I was thinking something like this 

````
export default function () {
  const { getWalletVM, getIsWalletVMChain } = storeToRefs(useWalletStore())

  const needsReconnect = computed(() => getWalletVM.value && !getIsWalletVMChain.value) // computed can be removed if it's not needed outside of this composable

  const doAfterReconnect = ({ onSuccess, onCancel }) => {
    if (needsReconnect.value) {
      openReconnectConnectWalletModal({
        onConnect: onSuccess,
        onCancel,
        closeAfterConnect: true,
      })
    }
    else {
      onSuccess()
    }
  }

  return {
    doAfterReconnect,
    needsReconnect
  }
}
````

this way you can remove `getWalletVM.value && !getIsWalletVMChain.value)` that was being duplicated

this 

````

  if (getWalletVM.value && !getIsWalletVMChain.value) {
    openReconnectWalletModal()
    return
  }
  emit('mint')

````

becomes 

`````
  useDoAfterReconnect({ onSuccess: () => {
      emit('mint')
  }})
`````


if you need more control you can create and use `needsReconnect` and `openReconnectConnectWalletModal` independently

**Comment by Jarsen136** [comment]
> > > @Jarsen136 dont remove the useReconnectModal use the openreconnectmodal there
> > 
> > 
> > Do you mean like this?
> > ```
> > const { openReconnectWalletModal } = useReconnectModal()
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I don't get the benefit or major difference of using it as a composable instead of utils function. Could you explain a bit? In the previous implementation, it was a composable function but with a different name.
> 
> sure, I was thinking something like this
> 
> ```
> export default function () {
>   const { getWalletVM, getIsWalletVMChain } = storeToRefs(useWalletStore())
> 
>   const needsReconnect = computed(() => getWalletVM.value && !getIsWalletVMChain.value) // computed can be removed if it's not needed outside of this composable
> 
>   const doAfterReconnect = ({ onSuccess, onCancel }) => {
>     if (needsReconnect.value) {
>       openReconnectWalletModal({
>         onConnect: onSuccess,
>         onCancel,
>         closeAfterConnect: true,
>       })
>     }
>     else {
>       onSuccess()
>     }
>   }
> 
>   return {
>     doAfterReconnect,
>     needsReconnect
>   }
> }
> ```
> 
> this way you can remove `getWalletVM.value && !getIsWalletVMChain.value)` that was being duplicated
> 
> this
> 
> ```
> 
>   if (getWalletVM.value && !getIsWalletVMChain.value) {
>     openReconnectWalletModal()
>     return
>   }
>   emit('mint')
> ```
> 
> becomes
> 
> ```
>   useDoAfterReconnect({ onSuccess: () => {
>       emit('mint')
>   }})
> ```
> 
> if you need more control you can create and use `needsReconnect` and `openReconnectWalletModal` independently


Now I understand your concern and you would like to reuse the check condition `getWalletVM.value && !getIsWalletVMChain.value)`. However, this check condition is not universal and does not apply to all cases.

// 1.  components/carousel/CarouselTypeDrops.vue
It checks if the chain of drop matches the walletVM.
image

// 2.  components/collection/drop/MintButton.vue
It checks if the current urlPrefix matches the walletVM.
image

In other words, the logic suitable for reuse is only `openReconnectWalletModal`. No matter putting it to the composable or utils function, it's fine. wdyt @hassnian

**Review by hassnian — APPROVED** [review]
